js/viewmodel.js

```diff
@@ -0,0 +1,221 @@
+window.FBE = window.FBE || {};
+
+(function (FBE) {
+    'use strict';
+    /* global Blueprint, TinyEmitter */
+
+    var events = new TinyEmitter(),
+        bp = new Blueprint(),
+        selectedItem,
+        entityNamesThatDoNotRotate = /beacon|roboport|lab|heat_pipe|pipe$|furnace|chest|pole|substation|solar|accumulator|centrifuge|wall|rocket_silo|radar|turret|speaker|power_switch|reactor|lamp|land_mine/i

```

Comment:
well, that be a piece of regex...

wouldn't an array and #indexOf work just as well?

Reply:
it'd be a much bigger array. With the regex I can say `chest` , whereas an array would need `wooden_chest`, `iron_chest`, etc. similar story for turrets, poles, furnaces.